### PR TITLE
Format phones + emails in chat

### PR DESCRIPTION
**Pull request by buoyad**
Format phone numbers + remove assertion suffixes from phones and emails in:
- [x] Inbox row
- [x] Desktop conversation header
- [x] Info panel

---
- [x] Improve header styling / long participants list handling (did some of this)
- [x] Look into parens special casing for NA numbers (special cased for +1)

### CODE REVIEW

**Comment by buoyad** [comment]
cc @keybase/y2ksquad this is ready

File: shared/chat/header.desktop.tsx [hunk 1]
```diff
@@ -85,17 +94,30 @@ const Header = (p: Props) => {
...
+                        <Kb.ConnectedUsernames
+                          colorFollowing={true}
+                          underline={true}
+                          inline={true}
+                          type="Header"
+                          usernames={[part]}
+                          onUsernameClicked="profile"
+                        />
+                      )}
+                      {i !== participants.length - 1 && <Kb.Text type="Header">, </Kb.Text>}
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/chat/header.desktop.tsx]: This is a local `participants` passed in by `map`. I'll change the name.

File: shared/constants/chat2/meta.tsx [hunk 2]
```diff
@@ -281,6 +282,21 @@ export const inboxUIItemToConversationMeta = (i: RPCChatTypes.InboxUIItem, allow
   let cannotWrite =
     i.convSettings && i.convSettings.minWriterRoleInfo ? i.convSettings.minWriterRoleInfo.cannotWrite : false
 
+  const participantToContactName = {}
+  const participantToDisplayName = {}
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/constants/chat2/meta.tsx]: We use contact names in the conversation header and the 'Share install link' banner

File: shared/chat/header.desktop.tsx [hunk 3]
```diff
@@ -85,17 +94,30 @@ const Header = (p: Props) => {
...
-                skipSelf={p.participants.length > 1 /* length ===1 means just you so show yourself */}
-              />
+            ) : withoutSelf ? (
+              <Kb.Box2 direction="horizontal" style={Styles.globalStyles.flexOne}>
+                <Kb.Text type="Header" lineClamp={1}>
+                  {withoutSelf.map((part, i, participants) => (
+                    <Kb.Text type="Header" key={part}>
+                      {p.displayNames[part] ? (
+                        <Kb.Text type="Header" onClick={() => p.onGoToProfile(part)}>
+                          {p.displayNames[part]}
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/chat/header.desktop.tsx]: I think I'll go with Mike's suggestion to factor it out into something shared between desktop + mobile

**Comment by buoyad** [comment]
Thanks @jzila, addressed comments PTAL

**Comment by buoyad** [comment]
@jzila this is ready again, I put the parsing into `Usernames` and removed the `displayName` map/plumbing.

File: shared/chat/header.desktop.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jzila) [reply shared/chat/header.desktop.tsx]: I see, but any reason this can't just be `withoutSelf`?

**Comment by buoyad** [comment]
Merging over `TestRotateRace` flake

**Pull request merged**